server/src/main/java/org/elasticsearch/indices/analysis/wrappers/StableApiWrappers.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices.analysis.wrappers;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.Tokenizer;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.indices.analysis.AnalysisModule;
+import org.elasticsearch.plugins.scanners.PluginInfo;
+import org.elasticsearch.plugins.scanners.StablePluginsRegistry;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A utility class containing methods that wraps the Stable plugin API with the old plugin api.
+ * Note that most old and stable api classes have the same names but differ in package name.
+ * Hence this class is avoiding imports and is using qualifying names
+ */
+public class StableApiWrappers {
```

Comment:
Are we aiming for stable plugins to always be wrapped, or will the new API migrate to be a first-class API in ES?

Reply:
Are we introducing top-level classes alongside the old plugin API to run stable plugins within ES, or is it all going through the existing `Plugin` API?

Reply:
new api will be wrapped to allow to insert into the existing data structures in `AnalysisModule` so we can reuse the existing internals